### ISSUE TITLE
HTTP2, request handling issue

### ISSUE THREAD

**Issue by neowutran** [actix/actix-web#438]
Using the StaticFile handler + http2, 
On a webpage like that https://neowutran.ovh/data/html/dps-783-3000.html
(Lot of differents pictures)

Only some pictures get loaded correctly (2-3 picture).
Other pictures got some weird issue, screenshot https://ibb.co/esy4ke 

With apache httpd: everything is loaded correctly
If I disable http2 with ``.no_http2()``, every resources are loaded correctly. 

What is loaded and what is not loaded is completly random. If you refresh the page, it will be some other pictures.

**Comment by fafhrd91** [comment]
could you send me files?

**Comment by neowutran** [comment]
Created a git with everything needed 
https://git.neowutran.ovh:8080/neowutran/Actix-web-issue-438 

modify the IP https://git.neowutran.ovh:8080/neowutran/Actix-web-issue-438/src/master/src/main.rs#L24

run: 
./target/debug/neowutran 8083

browse 
/data/html/dps-783-3000.html

To reproduce the issue you need TLS + HTTP2 and must be runned on a remote server. Was not able to reproduce the issue on localhost

**Comment by neowutran** [comment]
Tested to use wget mirroring, sometime I get thing like

```
Reusing existing connection to neowutran.ovh:443
HTTP request sent, awaiting response... 200 No headers, assuming HTTP/0.9
Length: unspecified
```

**Comment by neowutran** [comment]
( Can be reproduced with version 0.7.3, with the lastest change on master it will be http 1 only )

**Comment by neowutran** [comment]
Tested with master again, can be reproduced. 
( Rust-tls + HTTP2 )

**Comment by neowutran** [comment]
With (openssl + http2) instead of (rustls + http2) -> It is working as expected. 
So probably a bug with rustls

**Comment by fafhrd91** [comment]
I guess this is corruption bug

**Comment by ctz** [comment]
This was fixed in rustls 0.13.1 as well as tokio-rustls 0.7.2
